src/voter/mod.rs

```diff
@@ -385,7 +415,14 @@ impl<H, N, E: Environment<H, N>, GlobalIn, GlobalOut> Voter<H, N, E, GlobalIn, G
 								*last_finalized_number = finalized_number.clone();
 								self.env.finalize_block(finalized_hash, finalized_number, round_number, commit)?;
 							}
+							call_with(process_commit_outcome, CommitProcessingOutcome::Good);
+						} else {
+							// Failing validation of a commit is bad.
+							call_with(process_commit_outcome, CommitProcessingOutcome::Bad(commit.precommits.len()));
 						}
+					} else {
+						// Import to backgrounded round is good.
+						call_with(process_commit_outcome, CommitProcessingOutcome::Good);
```

Comment:
it might be helpful to find out how many of the messages we hadn't seen already. right now `round` ignores duplicates but we could change it to let us know when a vote was duplicate.